### PR TITLE
HPCC-16822 Extract query archive from DLL in WsWorkunits

### PR DESCRIPTION
**Pull request by wangkx**
Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

### CODE REVIEW

**Comment by wangkx** [comment]
@afishbeck please review.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 1]
```diff
@@ -508,7 +508,7 @@ void WsWuInfo::getHelpers(IEspECLWorkunit &info, unsigned flags)
 
             if (version > 1.30)
             {
-                info.setHasArchiveQuery(query->isArchive());
+                info.setHasArchiveQuery(query->isArchive() || query->hasArchive());
```
Comment (afishbeck) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: You don't need the query->isArchive() any more.

**Comment by afishbeck** [comment]
@wangkx one very minor comment, otherwise looks good.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (wangkx) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: Question: if hasArchive is not set, does query->hasArchive() return false?

**Comment by wangkx** [comment]
@richardkchapman this PR is ready to be merged.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (richardkchapman) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: Are you sure?

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 2]
```diff
@@ -2094,7 +2103,18 @@ void WsWuInfo::getWorkunitArchiveQuery(MemoryBuffer& buf)
     SCMStringBuffer queryText;
     query->getQueryText(queryText);
     if ((queryText.length() < 1) || !isArchiveQuery(queryText.str()))
-        throw MakeStringException(ECLWATCH_CANNOT_GET_WORKUNIT, "Archive Query not found for workunit %s.", wuid.str());
+    {
+        if (!query->hasArchive())
+            throw MakeStringException(ECLWATCH_CANNOT_GET_WORKUNIT, "Archive Query not found for workunit %s.", wuid.str());
```
Comment (richardkchapman) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: Don't really want the capital Q on these error texts...

**Comment by richardkchapman** [comment]
@wangkx One minor comment

**Comment by wangkx** [comment]
@richardkchapman The 'Archive Query' has been changed to 'Archive query' on the error texts.

**Pull request merged**